### PR TITLE
Adds draft phase functionality in legislation processes

### PR DESCRIPTION
**Pull request by jaflutz**
References
===================
#2930 - Legislation process draft phase

Objectives
===================
Adds a draft phase to the legislation process, and hides processes in this phase when enabled

Visual Changes
===================

screenshot-draft-phase

### CODE REVIEW

**Comment by decabeza** [comment]
Hi @jaflutz,

Thanks for the PR! 🙏 😌 

I leave you some comments on the code 😉

File: app/views/admin/legislation/processes/_form.html.erb [hunk 1]
```diff
@@ -17,6 +17,36 @@
 
   <% end %>
 
+  <div class="small-12 medium-4 column">
+    <label><%= t("admin.legislation.processes.form.draft_phase") %></label>
+    <p class="help-text"><%= t("admin.legislation.processes.form.draft_phase_description") %></p>
+  </div>
+
+  <div class="small-12 medium-3 column">
+    <%= f.label :draft_start_date, t("admin.legislation.processes.form.start") %>
```
Comment (decabeza) [inline app/views/admin/legislation/processes/_form.html.erb]: Could you delete this `<%= f.label...` and use: 🙏 

```erb
<%= f.text_field :draft_start_date,
                  label: t("admin.legislation.processes.form.start"),
                  value: format_date_for_calendar_form(@process.draft_start_date),
                  class: "js-calendar-full",
                  id: "draft_start_date" %>
```

File: app/views/admin/legislation/processes/_form.html.erb [hunk 2]
```diff
@@ -17,6 +17,36 @@
...
+    <%= f.text_field :draft_start_date,
+                      label: false,
+                      value: format_date_for_calendar_form(@process.draft_start_date),
+                      class: "js-calendar-full",
+                      id: "draft_start_date" %>
+  </div>
+
+  <div class="small-12 medium-3 column">
+    <%= f.label :draft_end_date, t("admin.legislation.processes.form.end") %>
```
Comment (decabeza) [inline app/views/admin/legislation/processes/_form.html.erb]: Could you delete this `<%= f.label...` and use: 🙏 

```erb
<%= f.text_field :draft_end_date,
                  label: t("admin.legislation.processes.form.end"),
                  value: format_date_for_calendar_form(@process.draft_end_date),
                  class: "js-calendar-full",
                  id: "draft_end_date" %>
```

File: app/views/admin/legislation/processes/_form.html.erb [hunk 3]
```diff
@@ -46,6 +76,7 @@
     <hr>
   </div>
 
+
```
Comment (decabeza) [inline app/views/admin/legislation/processes/_form.html.erb]: Please, remove unnecessary blank line 😉

File: spec/factories/legislations.rb [hunk 4]
```diff
@@ -21,12 +21,15 @@
...
     debate_end_date { Date.current + 2.days }
+    #draft_start_date { Date.current - 1.days }
+    #draft_end_date { Date.current + 1.days }
     draft_publication_date { Date.current - 1.day }
     allegations_start_date { Date.current }
     allegations_end_date { Date.current + 3.days }
     proposals_phase_start_date { Date.current }
     proposals_phase_end_date { Date.current + 2.days }
     result_publication_date { Date.current + 5.days }
+    #draft_phase_enabled true
```
Comment (decabeza) [inline spec/factories/legislations.rb]: Please, remove the three commented lines on this file 😉

File: spec/features/admin/legislation/processes_spec.rb [hunk 5]
```diff
@@ -75,6 +77,48 @@
...
+        click_link "Collaborative Legislation"
+      end
+
+      expect(page).not_to have_content 'An example legislation process'
+
+      click_link "New process"
+
+      fill_in 'legislation_process_title_en', with: 'An example legislation process in draft phase'
+      fill_in 'legislation_process_summary_en', with: 'Summary of the process'
+      fill_in 'legislation_process_description_en', with: 'Describing the process'
```
Comment (decabeza) [inline spec/features/admin/legislation/processes_spec.rb]: This spec it failing right now, could you change these lines to:

```
fill_in 'Process Title', with: 'An example legislation process in draft phase'
fill_in 'Summary', with: 'Summary of the process'
fill_in 'Description', with: 'Describing the process'
```

This change avoid spec fails 😉

File: config/locales/en/activerecord.yml [hunk 6]
```diff
@@ -340,6 +342,8 @@ en:
               invalid_date_range: must be on or after the start date
             debate_end_date:
               invalid_date_range: must be on or after the debate start date
+            draft_end_date:
+              invalid_date_range: must be on or after the draft start date
```
Comment (decabeza) [inline config/locales/en/activerecord.yml]: Could you include `locales/es/activerecord.yml` file with this: 🙏 

```
draft_start_date: Fecha de inicio del borrador
draft_end_date: Fecha de fin del borrador
```

```
draft_end_date:
  invalid_date_range: tiene que ser igual o posterior a la fecha de inicio del borrador
```

File: config/locales/en/admin.yml [hunk 7]
```diff
@@ -373,6 +373,8 @@ en:
           enabled: Enabled
           process: Process
           debate_phase: Debate phase
+          draft_phase: Draft phase
+          draft_phase_description: If this phase is active, the process won't be listed on processes index. Allow to preview the process and create content before the start.
```
Comment (decabeza) [inline config/locales/en/admin.yml]: Could you include `locales/es/admin.yml` file with this: 🙏 

```
draft_phase: Fase de borrador
draft_phase_description: Si esta fase está activa, el proceso no aparecerá en la página de procesos. Permite previsualizar el proceso y crear contenido antes del inicio.
```

File: spec/features/admin/legislation/processes_spec.rb [hunk 8]
```diff
@@ -75,6 +77,48 @@
       expect(page).to have_content 'Summary of the process'
       expect(page).not_to have_content 'Describing the process'
     end
+
```
Comment (decabeza) [inline spec/features/admin/legislation/processes_spec.rb]: Please, remove unnecessary blank line 😉

**Comment by jaflutz** [comment]
@decabeza and @voodoorai2000, thanks for reviewing the PR!

**Comment by voodoorai2000** [comment]
Thank you so much @jaflutz! Nice work 👌

**Pull request merged**